Allow tests to be disabled.

From: Thierry Reding <thierry.reding@avionic-design.de>

This patch adds a --disable-tests option to the configure script in
order to remove the build-dependency on gtk+-3.0.
---
 Makefile.am  |    6 +++++-
 configure.ac |   14 ++++++++++----
 2 files changed, 15 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 41ff37c..e13787a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,10 @@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = libnotify docs tools tests
+SUBDIRS = libnotify docs tools
+
+if ENABLE_TESTS
+SUBDIRS += tests
+endif
 
 pcdata_DATA = libnotify.pc
 pcdatadir = $(libdir)/pkgconfig
diff --git a/configure.ac b/configure.ac
index 31d068e..5c65e16 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,8 +93,14 @@ AC_SUBST(PACKAGE_LIBS)
 AC_SUBST([pkg_modules])
 PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
 
-tests_modules="gtk+-3.0 >= $REQ_GTK_VERSION"
-PKG_CHECK_MODULES(TESTS, [$tests_modules])
+AC_ARG_ENABLE(tests,
+	[AC_HELP_STRING([--disable-tests], [build tests (requires gtk+-3.0)])],
+	enable_tests=$enableval, enable_tests=yes)
+if test x$enable_tests = xyes; then
+	tests_modules="gtk+-3.0 >= $REQ_GTK_VERSION"
+	PKG_CHECK_MODULES(TESTS, [$tests_modules])
+fi
+AM_CONDITIONAL(ENABLE_TESTS, [test x$enable_tests = xyes])
 
 GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
 AC_SUBST(GLIB_GENMARSHAL)
